pdcupdater/utils.py

```diff
@@ -694,3 +696,23 @@ def inner(*args, **kwargs):
                     time.sleep(interval)
         return inner
     return wrapper
+
+
+def get_token(pdc_api_url, keytab):
+    """
+    Uses kerberos keytab for automatic authentication 
+    and retrieves the token from pdc-updater
+    """
+    if not os.path.exists(keytab):
+        raise OSError("Keytab file not found")
+
+    os.environ['KRB5_CLIENT_KTNAME'] = 'FILE:{0}'.format(keytab)
+    headers = {'Accept':'application/json'}
+    
+    try:
+        auth = requests_kerberos.HTTPKerberosAuth(force_preemptive=True)
+        url = '{0}/auth/token/obtain/'.format(pdc_api_url.rstrip('/'))
+        r = requests.get(url, headers=headers, auth=auth)
+    except requests.exceptions.RequestException:
+        log.exception("<<URL Error>>")
```

Comment:
I was thinking that since I am using log.exception, it will catch the exception and give a description. Hence, I just wrote the context.